taskvine/src/manager/vine_manager.c

```diff
@@ -2684,20 +2684,17 @@ static int vine_manager_transfer_capacity_available(struct vine_manager *q, stru
 
 		/* If not, then search for an available peer to provide it. */
 		/* Provide a substitute file object to describe the peer. */
-		if(m->file->type != VINE_MINI_TASK) {
-			HASH_TABLE_ITERATE(q->worker_table, id, peer){
-				if((remote_info = hash_table_lookup(peer->current_files, m->file->cached_name)) && remote_info->in_cache) {
-					char *peer_source =  string_format("worker://%s:%d/%s", peer->transfer_addr, peer->transfer_port, m->file->cached_name);
-					if(vine_current_transfers_source_in_use(q, peer_source) < q->worker_source_max_transfers) {	
-						m->substitute = vine_file_substitute_url(m->file,peer_source);
-						free(peer_source);
-						found_match = 1;
-						break;
-					} else {
-						free(peer_source);
-					}
-				}
-			}
+		if(m->file->type != VINE_MINI_TASK) 
+		{
+			vine_file_delete(m->substitute);
```

Comment:
This might go away with the separate bug fix, but it looks dangerous to delete without setting m->substitute to null.

Reply:
Yes that was an artifact from the previous implementation. I don't think it had any functional effect. I will remove it